PyPortal_UV_Index/code.py

```diff
@@ -0,0 +1,119 @@
+"""
+PyPortal UV Index display
+
+Adafruit invests time and resources providing this open source code.
+Please support Adafruit and open source hardware by purchasing
+products from Adafruit!
+
+Written by Dave Astels for Adafruit Industries
+Copyright (c) 2019 Adafruit Industries
+Licensed under the MIT license.
+
+All text above must be included in any redistribution.
+"""
+
+import time
+import json
+import board
+import displayio
+from adafruit_pyportal import PyPortal
+from adafruit_display_shapes.rect import Rect
+from adafruit_display_text.Label import Label
+from adafruit_bitmap_font import bitmap_font
+
+try:
+    from secrets import secrets
+except ImportError:
+    print("""WiFi settings are kept in secrets.py, please add them there!
+the secrets dictionary must contain 'ssid' and 'password' at a minimum""")
+    raise
+
+MAX_BAR_HEIGHT = 160
+MARGIN = 10
+SPACE_BETWEEN_BARS = 1
+
+COLORS = [0x00FF00, 0x83C602, 0xa2CF02,
+          0xF7DE03, 0xF6B502, 0xF78802,
+          0xF65201, 0xEA2709,
+          0xDA0115, 0xFC019E, 0xB548FF,
+          0x988FFE, 0x7EA7FE, 0x66BFFD, 0x4BD9FF]
+
+cwd = ("/"+__file__).rsplit('/', 1)[0]
+
+CAPTION_FONT_FILE = cwd+'/fonts/Helvetica-Bold-16.bdf'
+BAR_FONT_FILE = cwd+'/fonts/Arial-Bold-12.bdf'
+
+#pylint:disable=line-too-long
+url = 'https://enviro.epa.gov/enviro/efservice/getEnvirofactsUVHOURLY/ZIP/{0}/JSON'.format(secrets['zip'])
+#pylint:enable=line-too-long
+
+def extract_hour(date_time):
+    """Extract the hour in a format to use for display:
+    :param date_time: the timestamp from EPA UV readings
+    """
+    split_date_time = date_time.split()
+    hour = split_date_time[1]
+    suffix = split_date_time[2]
+    if hour[0] == '0':
+        hour = hour[1]
+    return '\n'.join([hour, suffix])
+
+def extract_date(date_time):
+    """Extract the date in a format to use for display:
+    :param date_time: the timestamp from EPA UV readings
+    """
+    return ' '.join(date_time.split('/')[0:2])
+
+pyportal = PyPortal(url=url,
+                    status_neopixel=board.NEOPIXEL,
+                    default_bg=0xFFFFFF,
+                    caption_font=CAPTION_FONT_FILE)
+
+canvas = displayio.Group(max_size=36)
+pyportal.splash.append(canvas)
+bar_font = bitmap_font.load_font(BAR_FONT_FILE)
+
+while True:
+    json_payload = ''
+    try:
+        json_payload = pyportal.fetch()
+        raw_data = json.loads(json_payload)
+    except (ValueError, RuntimeError) as ex:
+        print('Error: ', ex)
+        if isinstance(ex, ValueError):
+            print('JSON:', json_payload)
+        print('Retrying in 10 minutes')
+        time.sleep(600)
+        continue
+    data = [{'hour': extract_hour(d['DATE_TIME']), 'value': int(d['UV_VALUE'])}
```

Comment:
i like listcomps but they're a lil advanced, maybe use some temp variables for these entries so its a little easier to understand?